File: agents/core/AgentRegistry.kt
package com.antai.agents.core

class AgentRegistry {
    private val agents = mutableMapOf<String, Agent>()

    fun register(agent: Agent) {
        agents[agent.name] = agent
    }

    fun get(name: String): Agent? {
        return agents[name]
    }

    fun list(): List<String> {
        return agents.keys.toList()
    }
}

File: ANT_X_OS/core/agent_registry.py
from ant_common import Registry


class AgentRegistry:
    def __init__(self):
        self._registry = Registry()

    @property
    def agents(self):
        return self._registry.mapping

    def register(self, name, agent):
        self._registry.register(name, agent)

    def get(self, name):
        return self._registry.get(name)
